driver-legacy/src/main/com/mongodb/MongoClient.java

```diff
@@ -286,66 +355,69 @@ public MongoClient(final List<ServerAddress> seeds, final MongoCredential creden
      * @throws MongoException if theres a failure
      */
     public MongoClient(final MongoClientURI uri) {
-        this(createCluster(uri, null), uri.getOptions(), uri.getCredentials());
+        this(uri, null);
     }
 
     /**
      * Creates an instance described by a URI.
      *
      * <p>Note: Intended for driver and library authors to associate extra driver metadata with the connections.</p>
      *
-     * @param uri the URI
+     * @param uri                    the URI
      * @param mongoDriverInformation any driver information to associate with the MongoClient
      * @throws MongoException if theres a failure
      * @since 3.4
      */
-    public MongoClient(final MongoClientURI uri, final MongoDriverInformation mongoDriverInformation) {
-        this(createCluster(uri, mongoDriverInformation), uri.getOptions(), uri.getCredentials());
+    public MongoClient(final MongoClientURI uri, @Nullable final MongoDriverInformation mongoDriverInformation) {
+        this(uri.getOptions().asMongoClientSettings(
```

Comment:
This was the trickiest to get right and I had a number of bugs before (I think) handling everything properly, including:

* preserving SRV 
* preserving credential
* preserving MongoClientOptions

Reply:
Ack